### PR TITLE
feat: Add serviceMonitor resource

### PR DESCRIPTION
**Pull request by shubham-cmyk**
### Description
It would add a prometheus serviceMonitor resource.

### Check List
- [x] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [x] Helm chart version bumped
- [x] Helm chart `CHANGELOG.md` updated to reflect change

*PR sponsored by [Obmondo.com](https://obmondo.com)*

### CODE REVIEW

**Review by TheAlgo — DISMISSED** [review]
Thanks for raising the PR. Can you look at lint failures once? Also is there an Github issue for this feature? It will be great if we can have one.

**Comment by shubham-cmyk** [comment]
@TheAlgo 
Bumped the chart version can you run the check again

**Comment by eyenx** [comment]
We would love to have this feature in the chart!

**Comment by gaiksaya** [comment]
Hi @shubham-cmyk 
Can you please rebase? Looks like branch has a conflict.

**Comment by peterzhuamazon** [comment]
I will take a look tomorrow and see if I can help rebase.

Thanks.

**Comment by peterzhuamazon** [comment]
I updated the version and changelog and tried to push but not able to do so, seems like the repo belongs to an org:
```
$ git push origin serviceMonitor
ERROR: Permission to Obmondo/helm-charts-1.git denied to peterzhuamazon.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

Hi @shubham-cmyk could you please update the PR?

Thanks.

**Comment by VILJkid** [comment]
Hi team,

I'll be picking up where @shubham-cmyk left off to address the recent changes. I'll update the PR shortly with the necessary adjustments. Thanks for your patience, and I’ll keep you posted on the progress!

**Comment by eyenx** [comment]
Thanks @VILJkid

**Comment by VILJkid** [comment]
@eyenx, I’ve taken care of the rebase and chart version bump. @shubham-cmyk is on vacation, so please review the changes and inform me if any additional updates are needed.

**Review by eyenx — APPROVED** [review]

**Comment by eyenx** [comment]
@TheAlgo can we merge this?

File: charts/opensearch/values.yaml
Comment (prudhvigodithi) [inline charts/opensearch/values.yaml:533]: LGTM, thanks @shubham-cmyk, please take care of the following.

1) Update the chart readme with some details, better to mention to install the prometheus CRD's as prerequisite. 
2) Add a comment in the `values.yaml` file above the `serviceMonitor` line.
3) Please Update the `CHANGELOG.md` file. (Sample PR https://github.com/opensearch-project/helm-charts/pull/569).
4) This is a very good change can we have this to the OpenSearch Dashboards also?

I will go ahead and merge this PR once the above points are addressed.

Thank you 
@TheAlgo @getsaurabh02 @peterzhuamazon

Comment (peterzhuamazon) [reply charts/opensearch/values.yaml:533]: Agreed with @prudhvigodithi , awaiting contributor to address above before giving my approval as well.

Thanks.

Comment (VILJkid) [reply charts/opensearch/values.yaml:533]: Hi @prudhvigodithi and @peterzhuamazon 

Thank you so much for your feedback! 😊 I'm currently working on the requested changes.
Thanks again for your help and patience!

Comment (shubham-cmyk) [reply charts/opensearch/values.yaml:535]: We have pushed the required changes please continue to merge this :)

**Comment by VILJkid** [comment]
The requested changes have been pushed. Kindly review and merge this feature.

Thanks for your time,
@prudhvigodithi @eyenx @bbarani @DandyDeveloper @peterzhuamazon @gaiksaya @TheAlgo

**Comment by VILJkid** [comment]
Thanks @TheAlgo!

The lint failures, and changelog version comparison is fixed!

File: charts/opensearch-dashboards/CHANGELOG.md
Comment (peterzhuamazon) [inline charts/opensearch-dashboards/CHANGELOG.md:372]: This should be 2.21.0 as it is a feature not a patch, so need to bump the minor version imho.

Thanks.

File: charts/opensearch-dashboards/Chart.yaml
Comment (peterzhuamazon) [inline charts/opensearch-dashboards/Chart.yaml:18]: Same as above.

File: charts/opensearch/CHANGELOG.md
Comment (peterzhuamazon) [inline charts/opensearch/CHANGELOG.md:17]: Similar to above, bump minor version.

File: charts/opensearch/Chart.yaml
Comment (peterzhuamazon) [inline charts/opensearch/Chart.yaml:18]: Same as above.

File: charts/opensearch/CHANGELOG.md
Comment (peterzhuamazon) [inline charts/opensearch/CHANGELOG.md:454]: Same as above.

File: charts/opensearch-dashboards/CHANGELOG.md
Comment (peterzhuamazon) [inline charts/opensearch-dashboards/CHANGELOG.md:17]: Same as above.

**Comment by VILJkid** [comment]
Thanks @peterzhuamazon!

The fix has been pushed and the minor versions have been bumped up, as per the change request.

**Review by peterzhuamazon — APPROVED** [review]
Hi @shubham-cmyk @VILJkid ,

I am fine with this PR now.
Giving my approval.

Waiting for @prudhvigodithi @TheAlgo to do another pass before merging.
Once this is merged if you could help create another backport to 1.x branch, that would be great to have.

Thanks for contribution.

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Thanks @shubham-cmyk @VILJkid @peterzhuamazon @TheAlgo , LGTM we can merge once the CI checks pass.

**Comment by peterzhuamazon** [comment]
Thanks @shubham-cmyk @VILJkid ,

Would you be able to backport this change to 1.x branch?

Thanks!

**Comment by VILJkid** [comment]
Thanks @peterzhuamazon,

Yes, I'll raise another PR with this change for branch `1.x` shortly.

**Comment by VILJkid** [comment]
Hi @peterzhuamazon,

Here's the PR for backporting this change to `1.x` : https://github.com/opensearch-project/helm-charts/pull/578

**Pull request merged**
